base/abstractarraymath.jl

```diff
@@ -200,31 +200,61 @@ function repmat(a::AbstractVector, m::Int)
     return b
 end
 
-# Generalized repmat
-function repeat{T}(A::AbstractArray{T};
-                   inner::Array{Int} = ones(Int, ndims(A)),
-                   outer::Array{Int} = ones(Int, ndims(A)))
+"""
+    repeat(A::AbstractArray; inner=(), outer=())
```

Comment:
I don't think the additional complexity for the reader is worth it. `ones(Int, ndims(A))` is almost as clear.